system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -5354,6 +5370,8 @@ class CLdapClient : implements ILdapClient, public CInterface
         while(nextptr != NULL)
         {
             prevptr = nextptr + 2;
+            if (prevptr == nullptr || *prevptr == '\0')
+                break;
             nextptr = strstr(prevptr, "::");
         }
         if(prevptr != NULL && *prevptr != '\0')
```

Comment:
The problem coverity is reporting is that prevptr cannot be null at this point, so this check is redundant (and misleading).

It can either be resourcename - which cannot be null, or nextptr+2 which also cannot be null.